test/e2e/simple_test.go

```diff
@@ -0,0 +1,122 @@
+// +build e2e
+
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"flag"
+	"log"
+	"os"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	kuberrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/knative/build/pkg/apis/build/v1alpha1"
+	"github.com/knative/pkg/test"
+)
+
+// TestMain is called by the test binary generated by "go test", and is
+// responsible for setting up and tearing down the testing environment, namely
+// the test namespace.
+func TestMain(m *testing.M) {
+	flag.Parse()
+	clients, err := newClients(test.Flags.Kubeconfig, test.Flags.Cluster, buildTestNamespace)
+	if err != nil {
+		log.Fatalf("newClients: %v", err)
+	}
+
+	// Ensure the test namespace exists, by trying to create it and ignoring
+	// already-exists errors.
+	if _, err := clients.kubeClient.Kube.CoreV1().Namespaces().Create(&corev1.Namespace{
```

Comment:
This is a nice thing, we should adopt it for `serving` as well, instead of having to create a test space as a requirement.